python/src/aac/plugins/check/checkaac.aac

```diff
@@ -0,0 +1,20 @@
+plugin:
+  name: CheckAaC
+  package: aac.plugins.check
+  description: |
+    An AaC plugin that checks the quality of AaC file content based on defined constraints.
+  commands:
+    - name: check
+      help_text: |
+        Perform AaC file quality checks using defined constraints in the AaC models.
+      input:
+        - name: aac-file
+          type: file
+          description: The path to the architecture file you want to check against constraints.
+        - name: --fail-on-warn
```

Comment:
This flag relates to a previous task that was at the tail end of the previous iteration and was determined to be dropped since it was worked in this branch. So, the implementation needs to be evaluated against the following task: #644

Reply:
@crazynewidea thoughts on if this implementation met your plans for the flag to make it qualify for closure?

Reply:
yes, this should complete the fail-on-warn task #644